sled-agent/src/storage/dump_setup.rs

```diff
@@ -0,0 +1,153 @@
+use crate::storage_manager::DiskWrapper;
+use camino::Utf8PathBuf;
+use illumos_utils::zpool::ZpoolHealth;
+use omicron_common::disk::DiskIdentity;
+use sled_hardware::DiskVariant;
+use slog::Logger;
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::sync::MutexGuard;
+
+#[derive(Default)]
+pub struct DumpSetup {
+    savecore_lock: Arc<std::sync::Mutex<()>>,
+}
+
+impl DumpSetup {
+    pub(crate) async fn poll_dumpdev_setup(
+        &self,
+        disks: &mut MutexGuard<'_, HashMap<DiskIdentity, DiskWrapper>>,
+        log: Logger,
+    ) {
+        let mut dump_slices = Vec::new();
```

Comment:
Not a major issue, but it feels a little wasteful to call this and re-calculate all state for all disks when we're only actually inserting one disk at a time. We do have the `DumpSetup` struct; we could keep a cache of all the information we need?

(Not a blocker for this PR though, the only difference here is a minor efficiency difference)

Reply:
yeah, that's what i'm thinking for a cleaner follow-up, the thought behind what's here now is "get dumps configured at all, these machines can probably handle counting to 10 redundantly a few times at startup for now"